src/Lykke.Service.BlockchainWallets.Services/WalletService.cs

```diff
@@ -115,12 +141,41 @@ public class WalletService : IWalletService
                 || await AdditionalWalletExistsAsync(integrationLayerId, assetId, clientId);
         }
 
-        public async Task<(IEnumerable<IWallet>, string continuationToken)> GetClientWalletsAsync(Guid clientId, int take,
-            string continuationToken)
+        public async Task<(IEnumerable<WalletWithAddressExtensionDto>, string continuationToken)> GetClientWalletsAsync(Guid clientId, int take, string continuationToken)
         {
-            var (wallets, token) = await _walletRepository.TryGetForClientAsync(clientId, take, continuationToken);
+            var (wallets, token) = await _walletRepository.GetAllAsync(clientId, take, continuationToken);
+            
+            return 
+            (
+                await Task.WhenAll(wallets.Select(ConvertWalletToWalletWithAddressExtensionAsync)),
```

Comment:
Constrain number of parallel calls to avoid DDoS of http client

Reply:
Client has already been limited with `take` parameter. Moreover, client< in fact, can not have too much wallets, to take some additional actions.

Reply:
If constants and capabilities services will be thread safe, then first call to each blockchain will be cached and rest of calls will take results from the cache